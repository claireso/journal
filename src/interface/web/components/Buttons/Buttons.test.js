import { render } from '@testing-library/react'

import { ButtonPrimary } from './index'

describe('<ButtonPrimary />', () => {
  it('should render loader', () => {
    const { asFragment } = render(<ButtonPrimary isLoading>Button</ButtonPrimary>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render button', () => {
    const { asFragment } = render(<ButtonPrimary>Button</ButtonPrimary>)

    expect(asFragment()).toMatchSnapshot()
  })
})
