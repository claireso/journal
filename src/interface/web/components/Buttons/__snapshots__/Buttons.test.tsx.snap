// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPrimary /> should render button 1`] = `
<DocumentFragment>
  <button
    class="styles__1mdc28jh styles_layerStyle__1mdc28j0 styles_button_variant_primary__1mdc28jj styles_layerStyle__1mdc28j2 styles_button_size_md__1mdc28jn styles_layerStyle__1mdc28j6"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<ButtonPrimary /> should render loader 1`] = `
<DocumentFragment>
  <button
    class="styles__1mdc28jh styles_layerStyle__1mdc28j0 styles_button_variant_primary__1mdc28jj styles_layerStyle__1mdc28j2 styles_button_size_md__1mdc28jn styles_layerStyle__1mdc28j6 styles_button_loading_true__1mdc28jq styles_layerStyle__1mdc28j9"
  >
    <div
      class="styles__4787ng9 styles_layerStyle__4787ng1 styles_spinner_variant_currentcolor__4787ngc styles_layerStyle__4787ng4 styles_spinner_size_md__4787nge styles_layerStyle__4787ng6"
    />
  </button>
</DocumentFragment>
`;
